module/solutions/helper.js

```diff
@@ -935,20 +945,23 @@ module.exports = class SolutionsHelper {
 	static queryBasedOnRoleAndLocation(data, type = '') {
 		return new Promise(async (resolve, reject) => {
 			try {
-				let entities = []
+				let registryIds = []
 				let entityTypes = []
 				let filterQuery = {
 					isReusable: false,
 					isDeleted: false,
 				}
+				Object.keys(_.omit(data, ['role', 'filter', 'factor'])).forEach((key) => {
```

Comment:
@Prajwal17Tunerlabs you are not omiting type if it is present . That will cause issue. please add and test

Reply:
type keyword added.